great_expectations/rule_based_profiler/parameter_builder/parameter_builder.py

```diff
@@ -573,43 +580,38 @@ def _sanitize_metric_computation(
             for metric_value_idx in metric_value_indices:
                 metric_value: MetricValue = metric_values[metric_value_idx]
                 if enforce_numeric_metric:
-                    if isinstance(metric_value, (str, np.str_)):
-                        if not is_parseable_date(value=metric_value):
-                            raise ge_exceptions.ProfilerExecutionError(
-                                message=f"""Applicability of {self.__class__.__name__} is restricted to numeric-valued \
-and datetime-valued metrics (value {metric_value} of type "{str(type(metric_value))}" was computed).
-"""
+                    if pd.isnull(metric_value):
+                        if not replace_nan_with_zero:
+                            raise ValueError(
+                                f"""Computation of metric "{metric_name}" resulted in NaN ("not a number") value."""
                             )
+
+                        batch_metric_values.append(0.0)
                     elif not (
-                        isinstance(metric_value, datetime.datetime)
+                        (
+                            isinstance(metric_value, (str, np.str_))
+                            and is_parseable_date(value=metric_value)
+                        )
+                        or isinstance(metric_value, datetime.datetime)
                         or isinstance(metric_value, decimal.Decimal)
                         or np.issubdtype(metric_value.dtype, np.number)
                     ):
                         raise ge_exceptions.ProfilerExecutionError(
-                            message=f"""Applicability of {self.__class__.__name__} is restricted to numeric-valued \
-and datetime-valued metrics (value {metric_value} of type "{str(type(metric_value))}" was computed).
+                            message=f"""Applicability of {self.__class__.__name__} is restricted to numeric-valued and \
+datetime-valued metrics (value {metric_value} of type "{str(type(metric_value))}" was computed).
 """
                         )
-
-                    if pd.isnull(metric_value):
-                        if not replace_nan_with_zero:
-                            raise ValueError(
-                                f"""Computation of metric "{metric_name}" resulted in NaN ("not a number") value.
-"""
-                            )
-
-                        batch_metric_values.append(0.0)
                     else:
                         batch_metric_values.append(metric_value)
+                else:
+                    batch_metric_values.append(metric_value)
 
             metric_values_by_batch_id[batch_id] = batch_metric_values
 
         attributed_resolved_metrics.metric_values_by_batch_id = (
```

Comment:
What happens with this variable if we no longer return it?

Reply:
@NathanFarmer In this case, return is not needed; everything in the caller is "accepted as a side-effect" (the values are updated "in-situ").  Thanks!